File: Level_3/네트워크.py
def solution(n, computers):
    answer = len(computers)

    for i in range(len(computers)):
        for j in range(i+1, len(computers)):
            if computers[i][j] == 1:
                if computers[j][i] == 1:
                    answer -= 1

    return answer


n = 3
computers = [[1, 1, 0], [1, 1, 0], [0, 0, 1]]
print(solution(n, computers))

for m in range(2, 2):
    print('yes')

File: Level_3/네트워크.cpp
#include <string>
#include <vector>

using namespace std;

void dfs(int cur, vector<vector<int>>& computers, vector<bool>& visited, int n) {
    visited[cur] = true; // Set true cuurent visit
    for (int next = 0; next < n; next++) {
        if (computers[cur][next] == 1 && !visited[next]) {
            dfs(next, computers, visited, n); // Recursive
        }
    }
}

int solution(int n, vector<vector<int>> computers) {
    int answer = 0;
    vector<bool> visited(n, false);
    
    for (int i = 0; i < n; i++) {
        if (!visited[i]) {
            dfs(i, computers, visited, n);
            answer++; // Add connection to network
        }
    }
    
    return answer;
}